### DISCUSSION TITLE
Is this safe to set a component as a state value?

### DISCUSSION THREAD

**Discussion by ghost** [facebookexperimental/Recoil#1806] [Q&A] [answered]
For now, this code works fine. But need your advice from a performance and code structure perspective.

```javascript
const setPopup = useSetRecoilState(popupState);

const handleForgetPassword=()=>{
  setPopup(<ForgetPassword/>);
}
```

**Comment by drarmstr** [comment, marked as answer]
In general non-mutable objects should be used as state so state changes are properly captured.  It might make sense to use a handle or ID to reference components.
